components/common-go/log/log.go

```diff
@@ -131,28 +132,40 @@ func (f *gcpFormatter) Format(entry *log.Entry) ([]byte, error) {
 	return f.JSONFormatter.Format(entry)
 }
 
-// Writer produces a writer that wraps everything that's written to it in a log message.
+// JSONWriter produces a writer that wraps everything
+// that's written to it in a log message.
 // Callers are expected to close the returned writer.
 //
-// Beware: due to logEntry not being synchronised, writes to the returned writer must not
-//         concurrent.
-func Writer(logEntry *logrus.Entry) io.WriteCloser {
+// Beware: due to logEntry not being synchronised, writes
+// to the returned writer must not concurrent.
+func JSONWriter(logEntry *logrus.Entry) io.WriteCloser {
 	rd, rw := io.Pipe()
 	go func() {
-		b := make([]byte, 4096)
+		dec := json.NewDecoder(rd)
```

Comment:
we might want to add the following just in case
```go
defer func() {
    err := recover()
    if err != nil {
        logrus.Warn("log.JSONWriter panic: %v", err)
    }
}()
```

Reply:
done